tasks/rubies.yml

```diff
@@ -67,9 +67,9 @@
   changed_when: False
   failed_when: False
   register: detect_delete_ruby
-  when: rvm1_delete_ruby | bool
+  when: rvm1_delete_ruby is defined and rvm1_delete_ruby != ''
 
 - name: Delete ruby version
   command: '{{ rvm1_rvm }} remove {{ rvm1_delete_ruby }}'
   changed_when: False
```

Comment:
It would be better to report a change when the Ruby has been deleted.

I did a bit of testing and believe we can do that this way:

```ruby
- name: Delete ruby version
  command: '{{ rvm1_rvm }} remove {{ rvm1_delete_ruby }}'
  register: rvm_delete_command_result
  changed_when: "'#removing' in rvm_delete_command_result.stdout"
  when: rvm1_delete_ruby is defined and rvm1_delete_ruby != '' and detect_delete_ruby.rc == 0
```

@danochoa could you incorporate that maybe? If this is too hard to test on your side, let me know I can also work on it (but will need write access to your repo temporarily, or to issue a new PR & backport your commits).

Reply:
EDIT: I've updated the snippet, which contained a typo. This will need a bit more testing anyway!

Reply:
I've done more testing, and noticed `failed_when: False` (present before this PR) during the detection process. This will swallow errors occurring during the detection process.

We could instead use `rvm list strings`, fail if it fails (by default), then delete if the version is in the list.

I made a first attempt to have that behaviour, it would give:

```ruby
- name: Determine installed versions of Ruby (for deletion)
  command: '{{ rvm1_rvm }} list strings'
  changed_when: False
  register: rvm1_determine_installed_rubies
  # no need to run the call if nothing is to be deleted anyway
  when: rvm1_delete_ruby is defined and rvm1_delete_ruby != ''

- name: Delete ruby if relevant
  command: '{{ rvm1_rvm }} remove {{ rvm1_delete_ruby }}'
  register: rvm_delete_command_result
  changed_when: "'#removing' in rvm_delete_command_result.stdout"
  when:
    - rvm1_delete_ruby is defined and rvm1_delete_ruby != ''
    - rvm1_delete_ruby in (rvm1_determine_installed_rubies.stdout.splitlines())
```

Poke @rvm/ansible - there are other places where `failed_when: False` are used, and would be worth checking, just FYI.

Reply:
@thbar I appreciate the feedback. I think your suggested changes are maybe out of scope for this PR. But happy to include them since I agree with the improvements and the code you provided looks good and passed my quick tests. Thanks.

Reply:
@danochoa definitely agreed, it is out of scope strictly speaking, I hesitated a bit to suggest that! Thanks for the inclusion & testing. I'll approve the PR with final comments. Thank you!